sys/include/net/sock/tcp.h

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2016 Alexander Aring <aar@pengutronix.de>
+ *                    Freie Universität Berlin
+ *                    HAW Hamburg
+ *                    Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_sock_tcp    TCP sock API
+ * @ingroup     net_sock
+ * @brief       Sock submodule for TCP
+ * @{
+ *
+ * @file
+ * @brief   TCP sock definitions
+ *
+ * @author  Alexander Aring <aar@pengutronix.de>
+ * @author  Simon Brummer <simon.brummer@haw-hamburg.de>
+ * @author  Cenk Gündoğan <mail@cgundogan.de>
+ * @author  Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ * @author  Kaspar Schleiser <kaspar@schleiser.de>
+ */
+#ifndef NET_SOCK_TCP_H_
+#define NET_SOCK_TCP_H_
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/types.h>
+
+#include "net/sock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct _sock_tl_ep sock_tcp_ep_t;   /**< An end point for a TCP sock object */
```

Comment:
No I didn't, actually I wrote quite the opposite. ;)

``` sock_udp_ep_t ep = { .ip.family=AF_INET6, .ip.addr.ipv6=blah };```
``` sock_udp_ep_t ep = { .family=AF_INET6, .addr.ipv6=blah };```

Let's somehow go for the second, don't you think?